src/main/scala/com/gu/contentapi/models/AcastLog.scala

```diff
@@ -0,0 +1,188 @@
+   If the value of x-edge-result-type is Error and the value of x-edge-response-result-type is not Error, the client disconnected before finishing the download.
+
+
+x-edge-response-result-type
+----------------------------
+
+How CloudFront classified the response just before returning the response to the viewer. See also x-edge-result-type in field 14.
+
+Possible values include:
+
+ * Hit – CloudFront served the object to the viewer from the edge cache.
+ * RefreshHit – CloudFront found the object in the edge cache but it had expired, so CloudFront contacted the origin to verify that the cache has the latest version of the object.
+ * Miss – The request could not be satisfied by an object in the edge cache, so CloudFront forwarded the request to the origin server and returned the result to the viewer.
+ * LimitExceeded – The request was denied because a CloudFront limit was exceeded.
+ * CapacityExceeded – CloudFront returned a 503 error because the edge location didn't have enough capacity at the time of the request to serve the object.
+ * Error – Typically, this means the request resulted in a client error (sc-status is 4xx) or a server error (sc-status is 5xx).
+ * Redirect – CloudFront redirects from HTTP to HTTPS.
+
+If sc-status is 403 and you configured CloudFront to restrict the geographic distribution of your content, the request might have come from a restricted location. For more information about geo restriction, see Restricting the Geographic Distribution of Your Content.
+If the value of x-edge-result-type is Error and the value of x-edge-response-result-type is not Error, the client disconnected before finishing the download.
+
+
+x-edge-request-id
+-----------------
+An encrypted string that uniquely identifies a request.
+
+----------------
+| Other fields |
+----------------
+
+city
+country
+country code
+region code
+continent code
+dma code
+postal_code
+longitude
+latitude 
+
+*/
+
+case class AcastLog(
+  timestamp: String,
+  bytes: String,
+  range: String,
+  ipAddress: String,
+  request: String,
+  url: String, /* relative URL */
+  status: String,
+  referrer: String,
+  userAgent: String,
+  query: String,
+  host: String,
+  forwardedFor: String,
+  cloudfrontResultType: String,
+  cloudfrontResponseResultType: String,
+  cloudfrontRequestId: String,
+  city: String,
+  country: String,
+  countryCode: String,
+  region: String,
+  continentCode: String,
+  dmaCode: String,
+  postalCode: String,
+  longitude: String,
+  latitude: String
+)
+
+object AcastLog {
+
+  def apply(line: String): Option[AcastLog] = {
+
+    val log = CSVReader[AcastLog].readCSVFromString(line).headOption
+    val parsedLog = log collect { case Success(parsedLog) => parsedLog }
+    parsedLog.map(l => l.copy(userAgent = decode(l.userAgent)))
+  }
+
+  val decodingMap = ListMap(
```

Comment:
`ListMap` is map backed by a `List` implementation that preserves insertion order, such that iterators and traversal methods visit key-value pairs in the order whey were first inserted. This is needed because the decoding logic have strings that overlaps.